app/src/main/java/com/fbada/projectone/Car.kt

```diff
@@ -0,0 +1,349 @@
+     */
+    override fun navigate() {
+        if (insurance) {
+            if (dashboard == "big") {
+                price * 40
+                println("Beta is so easy to navigate")
+            } else {
+                price / 3
+                println("Beta map is too small")
+            }
+        } else {
+            println("This car needs insurance!")
+        }
+    }
+}
+/**
+ * Subclass Omega Model type car.  Extends BMW class.
+ * @property windowSize Size of windows
+ * @property dashboard Big or small dashboard type
+ * @property ecoRating Eco Rating Score
+ */
+class Omega(private val windowSize: Int, private val dashboard: String, val ecoRating: Int) : BMW("Petrol", "Manual") {
+    /**
+     * Navigation function which prints a message dependent on whether the dashboard is big or small.
+     */
+    override fun navigate() {
+        if (dashboard == "big") {
+            price * 20
+            println("Omega is so easy to navigate")
+        } else {
+            price / 13
+            println("Omega map is too small")
+        }
+    }
+}
+
+/**
+ * Subclass Mayback Model type car.  Extends Mercedes class.
+ */
+class Mayback : Mercedes(4, 3) {
+    /**
+     * @property bootSize Size of boot
+     * @property price Base price plus boot size
+     */
+    val bootSize: Double = 4.4
+    override var price: Double = super.price + bootSize
+}
+/**
+ * Subclass Sprinter Model type car.  Extends Mercedes class.
+ * @property rating Customer rating score.
+ * @property maxSpeed Maximum speed of vehicle.
+ * @constructor Creates a Sprinter with rating, max speed and gearbox and returns specific specifications
+ */
+open class Sprinter(private val rating: Int, val maxSpeed: Int) : Mercedes(2, 2) {
+    constructor(rating: Int, maxSpeed: Int, gearbox: String) : this(rating, maxSpeed) {
+        val specs =
+            "This Sprinter car has the following specs:\n Customer Rating:$rating\n Gearbox: $gearbox\n Max Speed:$maxSpeed"
+        println(specs)
+    }
+    /**
+     * @property price Base price plus Sprinter Rating Price and Sprinter Max Speed Price
+     */
+    override var price: Double = super.price + sprinterRatingPrice() + sprinterMaxSpeedPrice()
+    /**
+     * Calculates a price based on customer rating
+     */
+    fun sprinterRatingPrice(): Double {
+        val extras: Double = when (rating) {
+            in 0..10 -> 2.00 + rating
+            in 11..30 -> 6.00 + rating
+            in 31..55 -> 10.00 + rating
+            in 56..77 -> 15.00 + rating
+            else -> 22.00 + rating
+        }
+        return extras
+    }
+    /**
+     * Calculates a price based on vehicle max speed.
+     */
+    fun sprinterMaxSpeedPrice(): Double {
+        val extras: Double = when (maxSpeed) {
+            in 0..60 -> 10.00
+            in 61..80 -> 20.00
+            in 81..100 -> 30.00
+            else -> 50.00
+        }
+        return extras
+    }
```

Comment:
Why do we need the `extras` prop? Can't we `return` the when block directly?
Do you need to have those two methods publicly available? It seems to be an internal implementation detail.

Also both fun naming isn't obvious, could you rephrase it a bit to describe what exactly is happening. It should be exhaustive(like your doc). And in most cases if the naming is exhaustive you don't need to document it.

Reply:
All suggested improvements completed.